crates/oxc_semantic/src/reference.rs

```diff
@@ -36,28 +39,47 @@ impl Reference {
         self.symbol_id = Some(symbol_id);
     }
 
+    /// Returns `true` if the identifier value was read. This is not mutually
```

Comment:
How can this not be mutually exclusive? this enum is mutually exclusive, it's one or the other...

Reply:
It's not an enum, it's a bitflag. A reference can be both a read and a write,
e.g. `a += 1`

Reply:
why can't we just have `ReadWriteType::ReadAndWrite`? that feels objectively more clear?

Reply:
Wouldn't that make checking for read usages harder? You'd have to do something like

```rust
fn is_read(&self) -> bool {
  match self {
    Self::Read => true,
    Self::ReadWrite => true,
    _ => false
  }
}
```
Which seems more verbose to me and also can't be made into a const function

Reply:
This enum doesn’t have ReadWrite in it’s current form though

Reply:
It does - it's a bitflag, not an enum. This is produced using `ReferenceFlag::Read | ReferenceFlag::Write`